utils/hash_containers.c

```diff
@@ -102,14 +104,14 @@ retcode_t hash_set_add(hash_set_t *const set, flex_trit_t const *const hash) {
   return RC_OK;
 }
 
-retcode_t hash_set_append(hash_set_t const *const set1,
-                          hash_set_t *const set2) {
+retcode_t hash_set_append(hash_set_t const *const set_from,
+                          hash_set_t *const set_to) {
   retcode_t ret = RC_OK;
   hash_set_entry_t *iter = NULL, *tmp = NULL;
 
-  HASH_ITER(hh, *set1, iter, tmp) {
-    if (!hash_set_contains(set2, iter->hash)) {
-      if ((ret = hash_set_add(set2, iter->hash)) != RC_OK) {
+  HASH_ITER(hh, *set_from, iter, tmp) {
+    if (!hash_set_contains(set_to, iter->hash)) {
```

Comment:
If it's a set you shouldn't have to check if it contains it already

Reply:
`hash_set_add` could take a bool and set the value if it added or not ?

Reply:
don't think we care about whether it was added right now? but ya, could do that.

Reply:
I didn't see you were talking of `hash_set_append ` with the little piece of code.
You're right, but utash set does not check for unicity, we have to do it ourselves.
Anyway, this `hash_set_contains` should go to `hash_set_add` instead.